packages/grid/src/GridRenderer.ts

```diff
@@ -2503,6 +2557,109 @@ export class GridRenderer {
         handleWidth,
         hScrollBarSize - scrollBarCasingWidth
       );
+
+      if (!autoSelectRow && scrollBarSelectionTickColor != null) {
+        context.fillStyle = scrollBarSelectionTickColor;
+        // Scrollbar Selection Tick
+        const { selectedRanges, model, cursorColumn } = state;
+        const { visibleColumns, visibleColumnWidths } = metrics;
+        const { columnCount } = model;
+
+        const leftVisibleIndex = Math.min(...visibleColumns.values());
+        const rightVisibleIndex = Math.max(...visibleColumns.values());
+
+        const getTickX = (index: number) => {
+          if (index < leftVisibleIndex) {
+            return Math.round((index / columnCount) * barWidth);
+          }
+          if (index > rightVisibleIndex) {
+            const remainderRight =
+              (getOrThrow(visibleColumnXs, rightVisibleIndex) +
+                getOrThrow(visibleColumnWidths, rightVisibleIndex) -
+                barWidth) /
+              getOrThrow(visibleColumnWidths, rightVisibleIndex);
+
+            const rightTickSize =
+              (barWidth - (x + handleWidth)) /
+              (columnCount - rightVisibleIndex + remainderRight);
+
+            return Math.round(
+              barWidth - (columnCount - index + 1) * rightTickSize
+            );
+          }
+          return Math.round(
+            (getOrThrow(visibleColumnXs, index) / width) * handleWidth + x
+          );
+        };
+
+        const filteredRanges = [...selectedRanges].filter(
+          value => value.startColumn != null && value.endColumn != null
+        ) as NoneNullColumnRange[];
+
+        filteredRanges.sort(GridRenderer.compareColumns);
+
+        const mergedRanges: NoneNullColumnRange[] = [];
+
+        for (let i = 0; i < filteredRanges.length; i += 1) {
+          const range = filteredRanges[i];
+          const { startColumn, endColumn } = range;
+          if (i === 0) {
+            mergedRanges.push({ startColumn, endColumn });
+          } else if (
+            startColumn - 1 <=
+            mergedRanges[mergedRanges.length - 1].endColumn
+          ) {
+            mergedRanges[mergedRanges.length - 1].endColumn = Math.max(
+              mergedRanges[mergedRanges.length - 1].endColumn,
+              endColumn
+            );
+          } else {
+            mergedRanges.push({ startColumn, endColumn });
+          }
+        }
```

Comment:
Use `AxisRange`, and put something like this in `GridUtils` (similar to `consolidate` method in `GridRange`)